pkg/observer/pubsub.go

```diff
@@ -0,0 +1,229 @@
+package observer
+
+import (
+	"fmt"
+	"log/slog"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+const subscriberChannelBufferSize = 1000
+
+type Config struct {
+	SubscriberChannelBufferSize int64 `koanf:"subscriber_channel_buffer_size"`
+}
+
+var Pubsub *GoChannelPubsub
```

Comment:
Should this act as a singelton? If yes it is better to make this variable private and make a function to returns this like `GetPubSubInstance()` or something like this.